charts/producer-app-cleanup-job/templates/job.yaml

```
@@ -4,6 +4,12 @@ apiVersion: batch/v1
 kind: Job
 metadata:
   name: {{ template "producer-app-cleanup-job.fullname" . }}
+{{ - if .Values.annotations }}
```

```suggestion
{{- if .Values.annotations }}
```

does that cause the issues?